_posts/2019-11-13-Upgrading-a-Fairly-Large-Rails-Application-from-4.2-to-5.2.markdown

```diff
@@ -0,0 +1,25 @@
+---
+layout: post
+title: "Upgrading a Fairly Large Rails Application from 4.2 to 5.2"
+date: 2019-11-13 12:30:00
+categories: ["rails", "upgrades", "case-study"]
+author: "abizzinotto"
+---
+
+We recently collaborated with [Amitree](https://www.home.amitree.com/) on a [Rails upgrade](https://fastruby.io) for their application which "is an AI assistant for email, integrated deeply within Google and Microsoft’s products and is in heavy use by real estate professionals" and were able to speak to them about their experience with [Ombu Labs](https://www.ombulabs.com).
+
+<!--more-->
+
+We spoke with Marcus McLaughlin, Head of Engineering at [Amitree](https://www.home.amitree.com/), about the work performed to upgrade their app. Marcus states that their application is fairly large, about 5 years old and has a heavy reliance on background processing.
+
+[Amitree](https://www.home.amitree.com/) was having difficulty justifying developer cycles to the Rails upgrade project because they wanted to focus on growth as opposed to maintenance, a common issue experienced by many companies. Their application fell out of date because, in Marcus' words, "upgrading didn’t make the cut for our priority list, and we never had a business critical reason to upgrade the application until the version we were running was about to go entirely out of support and could result in security issues". That is why they decided to come to Ombu Labs for help.
+
+Initially, [Amitree](https://www.home.amitree.com/) ordered the Ombu Labs Roadmap so they could have a good estimate of what would be needed to do the upgrade. The roadmap proved to be useful "both in terms of the information that it provided, but also in terms of building my [Marcus'] confidence in Ombu Labs ability to deliver the estimate".
+
+We then executed a full upgrade of the application from Rails 4.2 to 5.2. This necessary upgrade of their application ensured their application continued running on a Rails version that is still supported, preventing security issues and bugs, something that is of critical importance for [Amitree](https://www.home.amitree.com/) because they prioritize the security and safety of user data.
```

Comment:
@ABizzinotto the second time you say `their application` sounds a bit repetitive. maybe you can replace it with something else?

Reply:
Done :)